play-json/jvm/src/test/scala/play/api/libs/json/JsonSpec.scala

```diff
@@ -477,6 +478,55 @@ class JsonSpec extends org.specs2.mutable.Specification {
       fromJson[JsonNode](jsNum).map(_.toString).must_==(JsSuccess("12.345"))
     }
 
+    "Serialize JsNumbers with integers correctly" in {
+      val numStrings = Seq(
+        "0",
+        "1",
+        "-1",
+        Int.MaxValue.toString,
+        Int.MinValue.toString,
+        Long.MaxValue.toString,
+        Long.MinValue.toString,
+        BigInteger.valueOf(Long.MaxValue).add(BigInteger.ONE).toString,
+        BigInteger.valueOf(Long.MinValue).add(BigInteger.valueOf(-1)).toString
+      )
+      numStrings.map { numString =>
+        val bigDec = new java.math.BigDecimal(numString)
+        Json.stringify(JsNumber(bigDec)).must_==(bigDec.toString)
+      }
+    }
+
+    "Serialize JsNumbers with decimal points correctly" in {
+      val numStrings = Seq(
+        "0.123",
+        "1.23456789",
+        "-1.23456789",
+        Float.MaxValue.toString,
+        Float.MinValue.toString,
+        Double.MaxValue.toString,
+        Double.MinValue.toString,
+        java.math.BigDecimal.valueOf(Double.MaxValue).add(java.math.BigDecimal.valueOf(1)).toString,
+        java.math.BigDecimal.valueOf(Double.MinValue).add(java.math.BigDecimal.valueOf(-1)).toString
+      )
+      numStrings.map { numString =>
```

Comment:
same

Reply:
fails with this when forced to check the strings are equal as opposed to the numbers are equal

```
[info]   + Serialize JsNumbers with integers correctly (14 ms)
[error]   x Serialize JsNumbers with decimal points correctly (10 ms)
[error]    '3.4028235E+38' != '3.4028235E38' (JsonSpec.scala:513)
[error]   x Serialize JsNumbers with e notation correctly (10 ms)
[error]    '1.23456789012345679012345679E+999' != '1.23456789012345679012345679e999' (JsonSpec.scala:526)
```

This also happens with #1077 which has these tests but not the main source change